core/lib/via_btc_client/examples/data_inscription_example.rs

```diff
@@ -0,0 +1,682 @@
+use inquire::ui::{Color, RenderConfig, StyleSheet, Styled};
+use inquire::Text;
+use std::os::unix::process;
+use std::str::FromStr;
+use std::vec;
+
+use bitcoin::hashes::Hash;
+use bitcoin::key::Keypair;
+use bitcoin::key::UntweakedPublicKey;
+use bitcoin::locktime::absolute;
+use bitcoin::opcodes::{all, OP_FALSE};
+use bitcoin::script::{Builder as ScriptBuilder, PushBytesBuf};
+use bitcoin::secp256k1::{Message, Secp256k1, SecretKey, Signing, Verification};
+use bitcoin::sighash::{EcdsaSighashType, Prevouts, SighashCache, TapSighashType};
+
+use bitcoin::taproot::{ControlBlock, LeafVersion, TaprootBuilder, TaprootSpendInfo};
+use bitcoin::{
+    transaction, Address, Amount, CompressedPublicKey, Network, OutPoint, PrivateKey, ScriptBuf,
+    Sequence, TapLeafHash, Transaction, TxIn, TxOut, Txid, WPubkeyHash, Witness,
+};
+
+use bitcoincore_rpc::RawTx;
+
+use serde_json::Value;
+
+
+#[tokio::main]
+async fn main() {
+    let secp = Secp256k1::new();
+
+    greeting();
```

Comment:
It's not "the rustiest" way to implement functionality. Instead of having multiple standalone functions like greeting(), get_user_input(), etc., consider organizing the functionality into a struct with associated methods
e.g.
```
struct InscriptionManager {
    secp: Secp256k1<All>,
}
impl InscriptionManager {
    fn new() -> Self {    }

    fn greet(&self) {    }

    fn get_user_input(&self) -> .. {    }
}
```